internal/executor/helm/commands.go

```diff
@@ -0,0 +1,79 @@
+package helm
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/alexflint/go-arg"
+	"github.com/mattn/go-shellwords"
+)
+
+// Commands defines all supported Helm plugin commands and their flags.
+type Commands struct {
+	Install  *InstallCommand  `arg:"subcommand:install"`
+	Version  *VersionCommand  `arg:"subcommand:version"`
+	Status   *StatusCommand   `arg:"subcommand:status"`
+	Test     *TestCommand     `arg:"subcommand:test"`
+	Rollback *RollbackCommand `arg:"subcommand:rollback"`
+	Upgrade  *UpgradeCommand  `arg:"subcommand:upgrade"`
+	Help     *HelpCommand     `arg:"subcommand:help"`
+
+	// embed on the root of the Command struct to inline all aliases.
+	UninstallCommandAliases
+	ListCommandAliases
+
+	GlobalFlags
+}
+
+// GlobalFlags holds flags supported by all Helm plugin commands
+type GlobalFlags struct {
+	Namespace  string `arg:"--namespace,-n"`
+	Debug      bool   `arg:"--debug"`
+	BurstLimit int    `arg:"--burst-limit"`
+}
+
+func parseRawCommand(rawCmd string) (Commands, []string, error) {
+	rawCmd = strings.TrimSpace(rawCmd)
+	if !strings.HasPrefix(rawCmd, PluginName) {
+		return Commands{}, nil, errors.New("the input command does not target the Helm plugin executor")
+	}
+	rawCmd = strings.TrimPrefix(rawCmd, PluginName)
+
+	var helmCmd Commands
+	p, err := arg.NewParser(arg.Config{}, &helmCmd)
+	if err != nil {
+		return helmCmd, nil, err
+	}
+
+	args, err := shellwords.Parse(rawCmd)
+	if err != nil {
+		return helmCmd, nil, err
+	}
+	err = p.Parse(renameVersionFlag(args))
+	if err != nil {
+		return helmCmd, nil, err
+	}
+
+	return helmCmd, args, nil
+}
+
+// The go-arg library is handling the `--version` flag internally, see:
+// https://github.com/alexflint/go-arg/blob/727f8533acca70ca429dce4bfea729a6af75c3f7/parse.go#L610
+//
+// In case of Helm the `--version` flag has a different purpose, so we just remove it for now.
+func renameVersionFlag(args []string) []string {
```

Comment:
Command `helm --version` causes the plugin to crash.

![image](https://user-images.githubusercontent.com/4786168/209125278-b05c7cb5-70e1-4f60-9ba1-88450fef924f.png)

Reply:
Fixed, I also added a new uni-test to make sure it won't happen again 👍